App.js

```diff
@@ -1,77 +1,25 @@
-  Text,
-  View,
-  Button,
-  TouchableWithoutFeedback,
-  ScrollView,
-} from "react-native";
+//import 'react-native-gesture-handler';
+import * as React from 'react';
+import { NavigationContainer } from '@react-navigation/native';
+import { createNativeStackNavigator } from '@react-navigation/native-stack';
 
-import Header from "./components/header";
-import Card from "./components/topCard";
-import MiddleCard from "./components/middleCard";
-import BottomCard from "./components/bottomCard";
+import HomeScreen from './components/HomeScreen';
+import MainScreen from './screens/MainScreen';
+import TabNavigation from './screens/TabNavigation';
+import SampleScreen2 from './screens/SampleScreen2';
+const Stack = createNativeStackNavigator();
 
-export default function App() {
-  const [totalWeight, setTotalWeight] = useState(1);
-  const [sheets, setSheets] = useState(0);
-  const [length, setLength] = useState(650);
-  const [width, setWidth] = useState(430);
-  const [gram, setGram] = useState(300);
-  const [selectedFormatId, setSelectedFormatId] = useState(1);
-  const [selectedSizeId, setSelectedSizeId] = useState(1);
-
-  const updateWeight = () => {
-    setTotalWeight(((length/1000) * (width/1000) * gram * sheets).toFixed(3));
-  };
-
-  const handleOnSizePress = (props) => {
-    setLength(props["dimension"]["length"]);
-    setWidth(props["dimension"]["width"]);
-  };
-  const handleOnQuantityChange = (props) => {
-    setSheets(props);
-  };
 
-  useEffect(updateWeight);
+export default function App() {
   return (
-    <TouchableWithoutFeedback onPress={Keyboard.dismiss}>
-      <ScrollView>
-        <View style={styles.container}>
-          <Header title="Paper Calculator" />
-          <Card
-            weight={totalWeight}
-            onChangeQuantity={handleOnQuantityChange}
-          />
-          <MiddleCard
-            onChangeFormat={setGram}
-            onChangeSize={handleOnSizePress}
-            selectedFormat={selectedFormatId}
-            onFormatPress={setSelectedFormatId}
-            selectedSize={selectedSizeId}
-            onSizePress={setSelectedSizeId}
-          />
-          <BottomCard
-            onChangeLength={setLength}
-            onChangeWidth={setWidth}
-            onChangeGrammage={setGram}
-            length={length}
-            width={width}
-            gram={gram}
-            onFormatPress={setSelectedFormatId}
-            onSizePress={setSelectedSizeId}
-          />
-        </View>
-      </ScrollView>
-    </TouchableWithoutFeedback>
+    <NavigationContainer>
+      <Stack.Navigator initialRouteName="Main" >
+        <Stack.Screen name="Main" component={MainScreen} />
+        <Stack.Screen name="Home" component={HomeScreen} />
+        <Stack.Screen name="TabNavigation" component={TabNavigation} options={{ title: 'Tab Navigation' }} />
+        <Stack.Screen name="SampleScreen2" component={SampleScreen2} options={{ title: 'Sample' }} />
+      </Stack.Navigator>
```

Comment:
Practice with adding all type of navigation on the same screen. You'll learn about nested navigation.